tests/omni_lock_rust/tests/test_otx.rs

```diff
@@ -1055,3 +1348,234 @@ fn test_cobuild_otx_random() {
+            tx_builder = tx_builder.output_data(e);
+        }
+        for e in otx.witnesses().into_iter() {
+            tx_builder = tx_builder.witness(e);
+        }
+    }
+
+    let tx = ckb_types::core::cell::resolve_transaction(tx_builder.build(), &mut HashSet::new(), &dl, &dl).unwrap();
+    let verifier = Verifier::default();
+    assert_script_error(verifier.verify(&tx, &dl).unwrap_err(), ERROR_FLOW);
+}
+
+#[test]
+fn test_cobuild_otx_double_input() {
+    let mut dl = Resource::default();
+    let mut px = Pickaxer::default();
+
+    let mut tx_builder = ckb_types::core::TransactionBuilder::default();
+    let os = schemas::basic::OtxStart::new_builder().build();
+    let wl = schemas::top_level::WitnessLayout::new_builder().set(os).build();
+    tx_builder = tx_builder.witness(wl.as_bytes().pack());
+    for otx in [generate_otx_a0(&mut dl, &mut px)] {
+        for e in otx.cell_deps_iter() {
+            tx_builder = tx_builder.cell_dep(e);
+        }
+        for e in otx.inputs().into_iter() {
+            tx_builder = tx_builder.input(e);
+        }
+        for e in otx.outputs().into_iter() {
+            tx_builder = tx_builder.output(e);
+        }
+        for e in otx.outputs_data().into_iter() {
+            tx_builder = tx_builder.output_data(e);
+        }
+        for e in otx.witnesses().into_iter() {
+            tx_builder = tx_builder.witness(e);
+        }
+    }
+    let os = schemas::basic::OtxStart::new_builder().build();
+    let wl = schemas::top_level::WitnessLayout::new_builder().set(os).build();
+    tx_builder = tx_builder.witness(wl.as_bytes().pack());
+    for otx in [generate_otx_a0(&mut dl, &mut px)] {
+        for e in otx.cell_deps_iter() {
+            tx_builder = tx_builder.cell_dep(e);
+        }
+        for e in otx.inputs().into_iter() {
+            tx_builder = tx_builder.input(e);
+        }
+        for e in otx.outputs().into_iter() {
+            tx_builder = tx_builder.output(e);
+        }
+        for e in otx.outputs_data().into_iter() {
+            tx_builder = tx_builder.output_data(e);
+        }
+        for e in otx.witnesses().into_iter() {
+            tx_builder = tx_builder.witness(e);
+        }
+    }
+
+    let tx = ckb_types::core::cell::resolve_transaction(tx_builder.build(), &mut HashSet::new(), &dl, &dl).unwrap();
+    let verifier = Verifier::default();
+    assert_script_error(verifier.verify(&tx, &dl).unwrap_err(), ERROR_OTX_START_DUP);
+}
+
+#[test]
+fn test_cobuild_otx_noexistent_type_script_hash() {
+    let mut dl = Resource::default();
+    let mut px = Pickaxer::default();
+    let tx = assemble_otx(vec![generate_otx_a3_fail(&mut dl, &mut px)]);
+    let tx = ckb_types::core::cell::resolve_transaction(tx, &mut HashSet::new(), &dl, &dl).unwrap();
+    let verifier = Verifier::default();
+    assert_script_error(verifier.verify(&tx, &dl).unwrap_err(), ERROR_TYPESCRIPT_MISSING);
+}
+
+#[test]
+fn test_cobuild_otx_msg_size_all_0() {
```

Comment:
test_cobuild_otx_xxx series functions can be simplified with "assemble_otx" function.